src/modules/guardian/guardian.service.ts

```diff
@@ -95,5 +96,48 @@ export class GuardianService {
       throw new Error(`Error fetching users with Guardian role: ${error.message}`);
     }
   }
+
+  //Checking if guardian has orphan
+  // async hasOrphans(guardianId: string): Promise<{ hasOrphans: boolean; count: number }> {
+  //   try {
+  //     const count = await this.prisma.orphan.count({
+  //       where: { userId: guardianId },
+  //     });
+  
+  //     return { hasOrphans: count > 0, count };
+  //   } catch (error) {
+  //     console.error(`Error checking orphans for guardian ${guardianId}: ${error.message}`);
+  //     throw new Error(`Error retrieving orphans for guardian with ID ${guardianId}.`);
+  //   }
+  // }
+
+  async hasOrphans(guardianId: string): Promise<{ hasOrphans: boolean }> {
+    try {
+      const userWithRoleAndOrphans = await this.prisma.user.findUnique({
+        where: { id: guardianId },
+        include: {
+          roles: true,
+          Orphan: true,
+        },
+      });
+  
+      if (!userWithRoleAndOrphans) {
+        throw new Error(`User with ID ${guardianId} not found.`);
+      }
+  
+      const isGuardian = userWithRoleAndOrphans.roles.some(role => role.roleName === UserType.GUARDIAN);
+  
+      if (!isGuardian) {
+        return { hasOrphans: false };
+      }
+  
+      const orphanCount = userWithRoleAndOrphans.Orphan.length;
+  
+      return { hasOrphans: orphanCount > 0};
+      } catch (error) {
+        console.error(`Error checking orphans for guardian ${guardianId}: ${error.message}`);
+        throw new Error(`Failed to retrieve orphans for guardian with ID ${guardianId}.`);
+      }
```

Comment:
good job!
will recommend you read more on when to use try and catch,
nice one dear... you can merge